projects/dashboards/src/test/dashboard-verification.ts

```diff
@@ -0,0 +1,32 @@
+import { NavigationService, TimeRangeService } from '@hypertrace/common';
+import { MetadataService } from '@hypertrace/distributed-tracing';
+import { GraphQlRequestService } from '@hypertrace/graphql-client';
+import { ModelJson } from '@hypertrace/hyperdash';
+import { DashboardManagerService } from '@hypertrace/hyperdash-angular';
+import { mockProvider, Spectator } from '@ngneat/spectator/jest';
+import { of } from 'rxjs';
+
+export const isValidModelJson = (
+  spectator: Spectator<unknown>,
+  definition: ModelJson | { json: ModelJson }
+): boolean => {
+  const json = 'json' in definition ? definition.json : definition;
+  try {
+    spectator.inject(DashboardManagerService).create(json);
+
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+export const mockDashboardProviders = [
```

Comment:
So, it's going to be a non-shallow test? Kind of makes sense for dashboards, if we want to check the correctness of the ModelJsons

Reply:
Yeah - I'd love for it to just provide the widgets, but the widgets bring along all kinds of services, some of which require mocking due to global config. Providing the widgets ourselves defeats the purpose of the test, so we're stuck with this.